app/trip/[id]/components/trip-details-tabs.tsx

```diff
@@ -28,51 +28,71 @@ type TripDetailsTabsProps = {
 export const TripDetailsTabs = ({ activities }: TripDetailsTabsProps) => {
   const borderColor = useColorModeValue('gray.300', 'gray.600')
 
-  // Sorting by date Time
-  const sortedActivities = activities.sort((a, b) => {
-    const dateA = new Date(a.timeFrom).getTime()
-    const dateB = new Date(b.timeFrom).getTime()
+  const sortedActivities = useMemo(() => {
```

Comment:
Nice use of useMemo to optimize performance by avoiding unnecessary recalculations!